vector/src/main/java/im/vector/app/features/settings/VectorSettingsPushRuleNotificationPreferenceFragment.kt

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.settings
+
+import androidx.lifecycle.lifecycleScope
+import androidx.preference.Preference
+import im.vector.app.core.preference.VectorCheckboxPreference
+import im.vector.app.core.utils.toast
+import kotlinx.coroutines.launch
+import org.matrix.android.sdk.api.pushrules.Action
+import org.matrix.android.sdk.api.pushrules.RuleIds
+import org.matrix.android.sdk.api.pushrules.RuleSetKey
+import org.matrix.android.sdk.api.pushrules.rest.PushRule
+import org.matrix.android.sdk.api.pushrules.rest.PushRuleAndKind
+
+abstract class VectorSettingsPushRuleNotificationPreferenceFragment
+    : VectorSettingsBaseFragment() {
+
+    /**
+     * @return the bing rule status boolean
+     */
+    fun ruleStatusIndexFor(ruleAndKind: PushRuleAndKind): Boolean {
```

Comment:
Please make all the `fun` private where it's possible, default visibility is public in Kotlin